### PR TITLE
backend: OAuth SASL extensions support

### PR DESCRIPTION
**Pull request by smaxx**
Kafka SASL OAuth supports sending extensions(custom key-value pairs) during OAuth authentication. These extensions are also supported by franz-go(https://pkg.go.dev/github.com/twmb/franz-go/pkg/sasl/oauth).
So this PR adds a support for these custom properties

### CODE REVIEW

File: backend/pkg/config/kafka_sasl_oauth.go [hunk 1]
```diff
@@ -28,6 +28,7 @@ type KafkaSASLOAuthBearer struct {
 	ClientSecret  string `yaml:"clientSecret"`
 	TokenEndpoint string `yaml:"tokenEndpoint"`
 	Scope         string `yaml:"scope"`
+	Extensions    string `yaml:"extensions"`
```
[the remark this answers is not in the document]
Comment (smaxx) [reply backend/pkg/config/kafka_sasl_oauth.go:31]: honestly i thought about this option as well(wouldn't require manual parsing as a major pro), however focused on the option which allows using as both yaml and command line flag.
Theoretically we could have 2nd hidden property which would be populated from a flag and then parsed only when we deal with a flag. What do you think? Or not to overcomplicate things just to support the yaml-based approach?

[the remark this answers is not in the document]
Comment (weeco) [reply backend/pkg/config/kafka_sasl_oauth.go:31]: imo we can just omit the flag parsing here and keep things simple.

[the remark this answers is not in the document]
Comment (smaxx) [reply backend/pkg/config/kafka_sasl_oauth.go]: please see the new version. Long story short: i tried introducing string-to-string map as you suggested, however Go parses map with lowercasing the keys(as far as i understand it is a well known issue in Go world). That is why not to depend on any case-related restrictions i fell back to collection of key-value structs. Works perfectly well

[the remark this answers is not in the document]
Comment (smaxx) [reply backend/pkg/config/kafka_sasl_oauth.go]: actually the lowercasing happens in backend/pkg/config/config.go(see around line 163), due to the way how we deal with yaml keys vs env variable names. So either the approach has to be reworked to support whatever possible key format or we need to leave the actual SASL extension key as a property value which by default is not modified during Yaml parsing process

[the remark this answers is not in the document]
Comment (smaxx) [reply backend/pkg/config/kafka_sasl_oauth.go]: @weeco please review if the current implementation is valid and makes sense to you. Thanks

**Review by weeco — APPROVED** [review]

**Comment by weeco** [comment]
Thanks for the contribution @smaxx , LGTM!

**Pull request merged**